### PR TITLE
Create compiler types - fixed p.2

### PR DESCRIPTION
**Pull request by giftedl**
Now supports vectors, potion, particle and sounds variables.
If you have no idea what this is, look at #44 

Theres also no more weird lines

### CODE REVIEW

**Comment by DeveloLongScript** [comment]
it broken nvm

**Comment by XTerPL** [comment]
you can commit to your branch to add the commit to the pull request smh

**Comment by homchom** [comment]
@DeveloLongScript yes, in the future please just add to the pre-existing pull request instead of making a new one (assuming it is the same "feature"). I am reopening this for convenience

**Comment by giftedl** [comment]
ALRIGHT, im sorry for the large inconvenience but it is now completely working. (maybe i should test 😕 )

please merge

im not very experienced with pull requests

File: src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class BaseVarCommand extends Command {
+    @Override
+    public void register(Minecraft mc, CommandDispatcher<FabricClientCommandSource> cd, CommandBuildContext context) {
+        cd.register(ArgBuilder.literal("basevar").executes(ctx -> {
+            ItemStack heldItem = mc.player.getMainHandItem();
+            CompoundTag tag = heldItem.getTagElement("PublicBukkitValues");
+
+            if (tag != null && tag.getString("hypercube:varitem") != "") {
+                    JsonObject data = JsonParser.parseString(tag.getString("hypercube:varitem")).getAsJsonObject();
+                    if (data.get("id").getAsString().contains("loc") || data.get("id").getAsString().contains("snd") || data.get("id").getAsString().contains("pot") || data.get("id").getAsString().contains("vec") || data.get("id").getAsString().contains("part")) {
```
Comment (tk2217) [inline src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java:31]: 1. Why restrict this, couldn't it work for any code item?
2. You seem to have misunderstood how identifiers work here, use `.equals`.

File: src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                        mc.keyboardHandler.setClipboard(Base64.getEncoder().encodeToString(data.get("data").getAsJsonObject().toString().getBytes()));
+                        ChatUtil.sendMessage("Successfully copied to the clipboard.", ChatType.SUCCESS);
+                    }
+                    else {
+                        ChatUtil.sendMessage("The item your holding is not a valid variable. (you can only use vectors, locations, potion vars, particle vars and sounds)", ChatType.FAIL);
```
Comment (tk2217) [inline src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java:36]: Don't duplicate code like this, it just clutters stuff.

**Review by tk2217 — CHANGES_REQUESTED** [review]
Code is okayish but iirc it's been mentioned that this might not be smth we want to toss in recode.

File: src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java (hunk 1 above)
Comment (giftedl) [reply src/main/java/io/github/homchom/recode/mod/commands/impl/other/BaseVarCommand.java:31]: I used .equals, and it didnt work. Also, there are some code items that can just be written in code, and don't really need to be every single code item.

**Comment by giftedl** [comment]
Im guessing no one cares about this anymore so I'm closing this.

**Pull request closed without merging**